### DISCUSSION TITLE
make `entry.server` and `entry.client` files optional

### DISCUSSION THREAD

**Discussion by mcansh** [remix-run/remix#4601] [Proposals]
make `entry.server` and `entry.client` files optional would allow for quicker demos and getting started from nothing easier without having to use the remix cli to bootstrap an entire app. just install the dependencies and make a `app/routes/index.tsx` file and you're done. plus most folks don't touch these files anyways so making them optional and not need to be on disk* is an added bonus (they'll still be on disk, but you know what i mean)

by default it would look for the files and fallback to ones we have stored in `@remix-run/dev`, which would be copies from https://github.com/remix-run/remix/tree/main/templates/remix

started a PR without making a proposal first https://github.com/remix-run/remix/pull/4600

**Comment by ryanflorence** [comment]
```
npx remix g entry.server.{tsx|jsx}
npx remix g entry.client.{tsx|jsx}
```

**Comment by aniravi24** [comment]
Will the templates still generate these files by default (or at least ask to generate them) or would they no longer be generated? Even the simplest production-ready remix app I want to build requires modifying every default file Remix comes with, so I'm not totally convinced that most people don't need these files. On top of that, now the dev server will have to check for these every time it runs. This seems analogous to create-react-app saying you don't need an `index.tsx` because users don't need to see `ReactDOM.createRoot`. I'm not strongly opinionated either way, but it generally looks like people are supportive of this change so I wanted to comment to better understand the rationale cause it sounds a little too magical and also seems like a minor optimization.

**Reply by mcansh** [reply]
imo i think they would (at least for now), this is more so for "how quickly can i get a remix app running" sort of thing. we have some CLI re-architecture coming up, so that could change, but i think we'd still output them by default

**Comment by MadadiSaeed** [comment]
I strongly disagree with this proposal for these reasons:
- Remix CLI is the fastest way to bootstrap a demo project instead of manually creating a project, since it installs and configures the project automatically.
- Every time you create a demo or production-ready project, you need at least one UI library to speed up the process and every react UI library needs to tweak one of these files or sometimes all of those.
- Even in some cases you need to change code in `entry.server` or `entry.client` to add your code. For example bot detection, configuring service worker and ... .
- Literally every time I put my global CSS configs in `root` file.

My Proposal: 

- Add an option to Remix starter to hide those files (default: generates them).
- Add a CLI command that generates those files.

**Reply by mcansh** [reply]
> Remix CLI is the fastest way to bootstrap a demo project instead of manually creating a project, since it installs and configures the project automagically.

CLI isn't going anywhere, it's just a new option if you want it

> Even in some cases you need to change code in entry.server or entry.client to add your code. For example bot detection, configuring service worker and ... .

we'll have an `eject`-esque command that'll copy one or the other (or both) to your local project where you can still do all that

> Literally every time I put my global CSS configs in root file.

forgot to update the title/description here, but root would still be required

**Comment by tombohub** [comment]
How do we create those files now? Whatever you do please update documentation too

**Reply by kiliman** [reply]
`npx remix reveal`

From v1.14.0 Release Notes
> ## Other notable changes
> `entry.server` and `entry.client` files are now optional. If excluded, Remix will use reasonable defaults at build-time. If you need customization for these files, you can run `npx remix reveal` and it will generate them for you.

https://github.com/remix-run/remix/releases/tag/remix%401.14.0

**Reply by tombohub** [reply]
Thank you
